### PR TITLE
First kernel-docs rewrite of comments

### PR DESCRIPTION
**Pull request by PeterMacleodThompson**
Updated maps.h using kernel-docs recommmendations.

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

### CODE REVIEW

**Comment by PeterMacleodThompson** [comment]
Hi Enric,

This is my 1st  attempt to apply kernel-docs to comments in hikingGPS programs. 
I started with maps.h which has the greatest need of update.   

Peter

File: src/include/maps.h [hunk 1]
```diff
@@ -1,104 +1,119 @@
...
+ * Peter Thompson Sept 2015
+ *
+ *
+ * a mapset is a set of maps that:
+ *    have the same scale, and the same "look and feel"
+ *    all maps can be stitched together to form a single huge map
+ *
+ * a mapindex is a list of pointers to all map images that make up 1 mapset.
+ *
+ */
```
Comment (eballetbo) [inline src/include/maps.h:13]: Good

File: src/include/maps.h [hunk 2]
```diff
@@ -1,104 +1,119 @@
...
+ *      C250k MAP HEIGHT = 1 degree 	6758 pixels varies somewhat map to map
+ *      C250K SIE 	80 gigbytes; 915 maps (du -h; ls -l | wc -l)
+ *	C250k map of Canada is 675m x 523m
+ *
+ *      CIMW  MAP WIDTH = 6 degrees		4127 pixels varies
+ *      CIMW  MAP HEIGHT = 4 degrees	3738 pixels varies
+ *      CIMW SIZE 	1.7 gigbytes; 78 maps (du -h; ls -l | wc -l)
+ *      CIMW map of Canada is 27m x 25m
+ *
 */
```
Comment (eballetbo) [inline src/include/maps.h:36]: Good

File: src/include/maps.h [hunk 3]
```diff
@@ -1,104 +1,119 @@
...
+ *	loaded from /MAPSPATH/CIMW/mapindex 
+ *       |-----------------------------------------  
+ *	 |dddmmddmm key	| mapfilename to load	
+ *       |----------------------------------------
+ *       |    ... approx 78 records ...		
+ *
+ *
+ *
+ */
+
```
Comment (eballetbo) [inline src/include/maps.h:83]: Good

File: src/include/maps.h [hunk 4]
```diff
@@ -1,104 +1,119 @@
...
+ * typedef struct PMTmapindex
+ *
+ * mapindex file contains list of all maps (filenames) in 1 mapset.
+ * One mapset per mapfolder.
+ * map files are located in folder in mapfolder name defined in PMTmapset.
+ * mapkey is dddmmddmm = longitude, latitude of NW corner of map.
+ * sorted by mapkey to enable binary search via linux:
+ * sort -k1n unsortedmapindex > mapindex
+ */
+typedef struct PMTmapindex { /* key = top left (NorthWest) corner of map  */
```
Comment (eballetbo) [inline src/include/maps.h:98]: This part should be something similar to this:

```
/**
 * typedef struct PMTmapindex - list of all maps (filenames) in 1 mapset.
 * @mapkey: format is dddmmddmm = longitude, latitude of NW corner of map. 
 * @mapwidth: format is ddmm = degrees*100+minutes 
 * @mapheight: format is ddmm = degrees*100+minutes 
 * @mapname: map filename
 *
 * There is one mapset per mapfolder sorted by mapkey to enable binary search
 * via linux:
 *
 *   sort -k1n unsortedmapindex > mapindex
 */
```

File: src/include/maps.h [hunk 5]
```diff
@@ -1,104 +1,119 @@
...
 
-/**************   PMTmapset   *****************************************/
 
-/*list of mapsets sorted by increasing kscale = order to be viewed */
+
+/**
+ * typedef struct PMTmapset
+ *
+ *list of mapsets sorted by increasing kscale = order to be viewed 
+ */
```
Comment (eballetbo) [inline src/include/maps.h:119]: Same as above

**Review by eballetbo — COMMENTED** [review]
Hi Peter,

Well done, only a mistake doing the comments for the structs. The format should be:
```
/**
 * typedef type_name - Brief description.
 * @arg1: description of arg1
 * @arg2: description of arg2
 *
 * Long description of the type.
 */
```

**Comment by PeterMacleodThompson** [comment]
Hi Enric,
Thanks for the excellent review!!   I will make those changes via an extra commit. 

On another note, clang-format with the new hikingGPS/src/.clang-format configuration file 
is giving errors:
eg: 
AlwaysBreakTemplateDeclarations: Multiline ==> must be binary so I changed to "false"
BreakInheritanceList: BeforeColon ==> unknown key (in clang-format version 6.0.0) 
so I commented it out  
... and there were 7 more unknown keys I commented out.  

The new .clang-format configuration file was added in commit 77b9568 
Is it ok for me to push my changes I made locally to .clang-format to github?

Ubuntu uses clang-format version 6.0.0 and Debian Stretch (which I am moving to)
uses version 3.8.1-24  (which may require some more changes).

The next commit of maps.h will have been run through clang-format.

Peter

**Comment by eballetbo** [comment]
Hi Peter,

I'd suggest to move to Debian Buster instead of Stretch, Stretch is the stable but old. Buster now marked as freeze, so will be the next stable. You can update from stretch to buster without problems. Also, Buster is what I'm using :-)

**Comment by PeterMacleodThompson** [comment]
Hi Enric,

Super!!   I will move to Debian Buster.   Life should be easier if we are on the same host.  

Peter

**Comment by PeterMacleodThompson** [comment]
Hi Enric,

I corrected the typedef sections in maps.h as you suggested.   I think the comments are pretty clear now.   I pushed another commit for a total of 2 commits for kernel-doc comments update on maps.h.   I did not create a second pull request.  I hope that is the correct workflow.   Let me know if we should be following a different workflow. 

Peter

**Comment by eballetbo** [comment]
I squashed the commit so only one commit is used to update the maps.h file. Patches merged.

**Comment by PeterMacleodThompson** [comment]
Super!   Thanks

Peter

**Pull request closed without merging**